### PR TITLE
Update zh-cn.json

### PR DESCRIPTION
**Pull request by steve02081504**

### CODE REVIEW

**Comment by Wolfsblvt** [comment]
I feel like we should find a better way to get translatable strings for the buttons than manually adding them hidden to the already existing DOM.

If we go this way, there is no reason why we'd have to do custom buttons for popups really.

Also don't feel like adding translation properties to the actual code-behind classes that have to be provided with all the code calls of popups is a good choice either.

Don't have a good solution yet, but let me think about this.

File: public/scripts/templates/charTagImport.html [hunk 1]
```diff
@@ -1,4 +1,4 @@
-<h3>Import Tags For {{charName}}</h3>
+<h3><span data-i18n="Import Tags For _begin">Import Tags For</span>{{charName}}<span data-i18n="Import Tags For _end"></span></h3>
```
Comment (Wolfsblvt) [inline public/scripts/templates/charTagImport.html]: This breaks layout for any space-reliant language.
image

File: public/scripts/templates/charTagImport.html [hunk 2]
```diff
@@ -23,13 +23,19 @@ <h4 data-i18n="Folder Tags">Folder Tags</h4>
...
         <label class="checkbox flex-container alignitemscenter flexNoGap m-t-3" for="import_remember_option">
             <input type="checkbox" id="import_remember_option" name="import_remember_option" />
-            <span data-i18n="Remember my choice">
-                Remember my choice
-                <div class="fa-solid fa-circle-info opacity50p"
-                    data-i18n="[title]Remember the chosen import option&#010;If anything besides 'Cancel' is selected, this dialog will not show up anymore.&#010;To change this, go to the settings and modify &quot;Tag Import Option&quot;.&#010;&#010;If the &quot;Import&quot; option is chosen, the global setting will stay on &quot;Ask&quot;."
-                    title="Remember the chosen import option&#010;If anything besides 'Cancel' is selected, this dialog will not show up anymore.&#010;To change this, go to the settings and modify &quot;Tag Import Option&quot;.&#010;&#010;If the &quot;Import&quot; option is chosen, the global setting will stay on &quot;Ask&quot;.">
-                </div>
-            </span>
+            <span data-i18n="Remember my choice">Remember my choice</span>
```
Comment (Wolfsblvt) [inline public/scripts/templates/charTagImport.html]: Closing the span before the fa icon isn't what is usually used in ST.  
Breaks the formatting of the icon inline a bit at least, making weird alignment.  
Below is this PR, top is on staging.
image

File: public/scripts/popup.js [hunk 3]
```diff
@@ -138,7 +140,12 @@ export class Popup {
         if (allowVerticalScrolling) this.dlg.classList.add('vertical_scrolling_dialogue_popup');
 
         // If custom button captions are provided, we set them beforehand
-        this.okButton.textContent = typeof okButton === 'string' ? okButton : 'OK';
+        if (okButton) {
```
Comment (Wolfsblvt) [inline public/scripts/popup.js]: This `if` prevents the fault text of `OK` from being inserted into the button by default, if not defined by the function call.  
Don't ask me why it's "Cancel" in the HTML, and default to "OK" in the js code, but we shouldn't break it, or many popups would look wrong.

File: public/scripts/popup.js [hunk 4]
```diff
@@ -41,6 +42,7 @@ export const POPUP_RESULT = {
 /**
  * @typedef {object} CustomPopupButton
  * @property {string} text - The text of the button
+ * @property {string} text_i18n - The i18n key of the button
```
Comment (Wolfsblvt) [inline public/scripts/popup.js]: Forcing a translation property onto every button a dev might design is not a good design. At least imho. Either use the actual button text as a translation base value, or there would actually have to be some kind of selector functionality to "find" the contents to translate.

With code changes like this, you blow up actual development code and contributor stuff with needed translation values.

Generally, I think translation is a reasonable thing. But it should be an *afterthought*, coming after the actual devving of features. If it's breaking code, or making coding more complicated, its a kind of trade-off that I am not sure is worth it. Personally.

**Comment by Cohee1207** [comment]
Please revert changes related to buttons. You can leave the rest (HTML attributes).

**Comment by steve02081504** [comment]
> Please revert changes related to buttons. You can leave the rest (HTML attributes).

fixed

File: public/scripts/templates/charTagImport.html (hunk 2 above)
Comment (Wolfsblvt) [reply public/scripts/templates/charTagImport.html]: Still an issue. Small one though. I think you can just revert to the older, big <span> tag again and add a small one just for the text with your il18n attribute.

File: public/scripts/templates/charTagImport.html (hunk 1 above)
Comment (Wolfsblvt) [reply public/scripts/templates/charTagImport.html]: Also still here. You *might* be able to solve this by making the text heading a flexbox and then using `justify-content: space-between;`, but I am not sure.

**Review by Wolfsblvt — CHANGES_REQUESTED** [review]
Still visually breaks a few things that should be dealt with first, I figure.
@Cohee1207 see my review comments.

File: public/scripts/templates/charTagImport.html (hunk 1 above)
Comment (Cohee1207) [reply public/scripts/templates/charTagImport.html]: Or `&nbsp;` before the macro. Adds a space for every language, but it's a lesser evil.

Comment (steve02081504) [reply public/scripts/templates/charTagImport.html]: Some languages may not want this space to exist so
Fixed in [`017cebb` (#2411)](https://github.com/SillyTavern/SillyTavern/pull/2411/commits/017cebb312e62f6c7de48ccc58d56454171c9def).

Comment (Wolfsblvt) [reply public/scripts/templates/charTagImport.html]: That's actually clever. That works.

File: public/scripts/templates/charTagImport.html (hunk 2 above)
Comment (steve02081504) [reply public/scripts/templates/charTagImport.html]: may fixed in [`8117d9a` (#2411)](https://github.com/SillyTavern/SillyTavern/pull/2411/commits/8117d9aafd285a6d3219606f7865dfbd114efefa) ..?

File: public/scripts/popup.js (hunk 3 above)
Comment (steve02081504) [reply public/scripts/popup.js]: I'm not sure why I'm seeing this review two days later...
Sorry if it put you in a bad mood.😢

File: public/scripts/popup.js (hunk 4 above)
Comment (steve02081504) [reply public/scripts/popup.js]: as above, I'm not sure why I'm seeing this review two days later...
Sorry if it put you in a bad mood, again.😢

File: public/scripts/popup.js (hunk 3 above)
Comment (Wolfsblvt) [reply public/scripts/popup.js]: All good! You didn't, it's oki.

**Review by Wolfsblvt — APPROVED** [review]
All good now.

**Pull request merged**